### PR TITLE
Add AirflowSecretsTomlProvider

### PR DESCRIPTION
**Pull request by burnash**
This pull request implements AirflowSecretsTomlProvider. This enables users to store their secrets in Airflow variables, making it easier to manage their configurations. 

Implements #223

### CODE REVIEW

**Review by rudolfix — COMMENTED** [review]
Looks good! what we need are tests. you can take a look at #190 it seems that we need to add airflow to our dev dependencies and then you can just add this variable to in-memory metadata database https://stackoverflow.com/questions/69289343/can-you-create-a-variable-directly-in-an-airflow-dag

so it should be trivial

**Comment by burnash** [comment]
@rudolfix the tests pass now with the updated poetry.lock. So it's ready for review.

File: dlt/common/configuration/specs/config_providers_context.py [hunk 1]
```diff
@@ -40,7 +39,51 @@ def add_provider(self, provider: ConfigProvider) -> None:
 
     @staticmethod
     def initial_providers() -> List[ConfigProvider]:
-        return [EnvironProvider(), SecretsTomlProvider(add_global_config=True), ConfigTomlProvider(add_global_config=True)]
+        providers = [
+            EnvironProvider(),
+            SecretsTomlProvider(add_global_config=True),
+            ConfigTomlProvider(add_global_config=True)
+        ]
+        providers += _extra_providers()
```
Comment (burnash) [inline dlt/common/configuration/specs/config_providers_context.py:47]: @rudolfix I wanted to decouple airflow provider from ConfigProvidersContext in a simples way possible. 
Later, if needed, we can implement this with a proper plugin-based architecture.

**Review by rudolfix — CHANGES_REQUESTED** [review]
@burnash just one small question to the workflow cache. otherwise it looks excellent. thanks for all the tests!

File: .github/workflows/test_airflow.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}-dbt-runner
```
[the remark this answers is not in the document]
Comment (burnash) [reply .github/workflows/test_airflow.yml]: Fixed

**Review by rudolfix — APPROVED** [review]

**Pull request merged**